### PR TITLE
23 Pattern library / Style guide base

### PR DESCRIPTION
**Pull request by Krank3n**
# Includes tasks

[23 Pattern library / Style guide base](https://trello.com/c/JCGpwaIn/23-pattern-library-style-guide)

# Fulfills

- [x] Install Storybook
- [x] Install CoreUI dependencies
- [x] Setup project to includes and build our theme scss

### CODE REVIEW

File: src/index.stories.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export const withText = () => <Button>Hello Button</Button>;
+
+export const withEmoji = () => (
+  <Button>
+    <span role="img" aria-label="so cool">
+      😀 😎 👍 💯
+    </span>
+  </Button>
+);
```
Comment (jakxnz) [inline src/index.stories.js:15]: Do we need this demo stuff in here?

File: src/js/app.js [hunk 2]
```diff
@@ -29,9 +29,9 @@ const App = () => (
   <BrowserRouter>
     <ApolloProvider client={client}>
       <IntlProvider locale={locale} message={t[locale]}>
-        <Switch>
-          <Route path="/" component={Home} />
-        </Switch>
+          <Switch>
+            <Route path="/" component={Home} />
+          </Switch>
```
Comment (jakxnz) [inline src/js/app.js:34]: Odd. It doesn't look like this needed indenting?

File: src/js/components/Header/Header.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    return (
+      <>
+        <h2>{ brandName }</h2>
+        <button type="button" onClick={() => changeBrandName('New name')}>Change brand name</button>
+      </>
+    );
+  }
+}
+
+export default Header;
```
Comment (jakxnz) [inline src/js/components/Header/Header.js:18]: This doesn't need to be part of this PR does it?

File: src/js/components/Home/Home.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import { FormattedMessage } from 'react-intl';
+import Buttons from '../Patterns/Buttons/Buttons';
```
Comment (jakxnz) [inline src/js/components/Home/Home.js:3]: This doesn't need to be part of this PR does it?

File: src/js/components/Home/Home.js [hunk 5]
```diff
@@ -15,6 +16,7 @@ const Home = () => (
         defaultMessage="Live your best adventure lifestyle"
       />
     </p>
+    <Buttons />
```
Comment (jakxnz) [inline src/js/components/Home/Home.js:19]: This doesn't need to be part of this PR does it?

File: src/js/components/Patterns/Buttons/Buttons.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+          <button type="button" className="btn btn-dark"><span className="cil-contrast btn-icon mr-2"></span> Dark
+          </button>
+          <button type="button" className="btn btn-link">Link</button>
+        </div>
+      </div>
+    </div>
+  </>
+);
+
+export default Buttons;
```
Comment (jakxnz) [inline src/js/components/Patterns/Buttons/Buttons.js:33]: This doesn't need to be part of this PR does it?

File: src/scss/_sidarktheme.scss [hunk 7]
```diff
@@ -0,0 +1,284 @@
+// Variable overrides
```
Comment (jakxnz) [inline src/scss/_sidarktheme.scss:1]: Do we need to name something "si" when it's in the Shred Index project?

File: src/toms.stories.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+          <button type="button" className="btn btn-light"><span className="cil-contrast btn-icon mr-2"></span> Light
+          </button>
+          <button type="button" className="btn btn-dark"><span className="cil-contrast btn-icon mr-2"></span> Dark
+          </button>
+          <button type="button" className="btn btn-link">Link</button>
+        </div>
+      </div>
+    </div>
+  </>
+);
```
Comment (jakxnz) [inline src/toms.stories.js:34]: This doesn't need to be part of this PR does it?

File: .gitignore [hunk 9]
```diff
@@ -3,4 +3,6 @@ public/dist
 public/patterns.html
 infrastructure
 .env
-coverage
\ No newline at end of file
+coverage
+.idea
+.circleci
```
Comment (jakxnz) [inline .gitignore:8]: We definitely don't want to git ignore the `.circleci` folder

File: src/dev/patterns.html [hunk 10]
```diff
@@ -8,7 +8,7 @@
   <meta name="description" content="ShredIndex - Pattern library">
   <meta name="author" content="Underflip">
 
-  <link rel="stylesheet" href="../css/style.css">
+  <link rel="stylesheet" href="dist/css/style.css?v=1.0">
```
Comment (jakxnz) [inline src/dev/patterns.html:11]: We don't need this file anymore aye?

File: src/js/app.js (hunk 2 above)
Comment (jakxnz) [reply src/js/app.js:34]: I think you may have changed this to 4 space indentation. It should be 2 right?

File: src/stories/buttons.stories.js [hunk 11]
```diff
@@ -0,0 +1,1540 @@
...
+          <CCard>
+            <CCardHeader>
+              Brand button
+            </CCardHeader>
+            <CCardBody>
+              <h6>Size Small
+                <small> Add this class <code>.btn-sm</code></small>
+              </h6>
+              <p>
+                <CButton size="sm" className="btn-facebook btn-brand mr-1 mb-1"><CIcon
```
Comment (jakxnz) [inline src/stories/buttons.stories.js:1298]: As far as I can tell, <CIcon name=cib-facebook"/> should work. I had a quick look but couldn't see anything obvious. I'm assuming we're missing something? It's a shame the way that you've done Icons in the icon's story is different from how they're done here.

File: src/js/app.js [hunk 12]
```diff
@@ -1,11 +1,11 @@
...
+import {ApolloProvider} from '@apollo/react-hooks';
+import {InMemoryCache} from 'apollo-cache-inmemory';
+import {ApolloClient} from 'apollo-client';
+import {HttpLink} from 'apollo-link-http';
 import React from 'react';
 import ReactDOM from 'react-dom';
-import { IntlProvider } from 'react-intl';
-import { BrowserRouter, Route, Switch } from 'react-router-dom';
+import {IntlProvider} from 'react-intl';
+import {BrowserRouter, Route, Switch} from 'react-router-dom';
```
[the remark this answers is not in the document]
Comment (Krank3n) [reply src/js/app.js:8]: My linting is still doing some weird stuff, still haven't spent the time on trying to sort it

File: src/stories/index.stories.js [hunk 13]
```diff
@@ -0,0 +1,190 @@
+import React from 'react';
+
+import '../../.storybook/storybook.scss';
+
+export default { title: 'Typography' };
+
+export const Typography = () => {
```
[the remark this answers is not in the document]
Comment (Krank3n) [reply src/stories/index.stories.js:7]: Yea I'll do this... but then this file will just be for importing the SCSS

[the remark this answers is not in the document]
Comment (jakxnz) [reply src/stories/index.stories.js:7]: Yep, that's all good.

File: src/stories/buttons.stories.js (hunk 11 above)
Comment (Krank3n) [reply src/stories/buttons.stories.js:1298]: Yea, Both ways are the same as what they have done in CoreUI Admin

File: src/js/components/Header/header.stories.js [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+import {
+  CHeader,
+  CHeaderNav,
+  CHeaderNavItem,
+  CHeaderNavLink,
+} from '@coreui/react';
+
+export default { title: 'Header' };
+
+export const Header = () => (
```
Comment (jakxnz) [inline src/js/components/Header/header.stories.js:11]: Is this header component in-scope for this PR? Or can we remove it?

File: src/stories/buttons.stories.js [hunk 15]
```diff
@@ -0,0 +1,2274 @@
...
+                  name="cib-reddit"
+                />
+              </CButton>
+              <CButton size="lg" className="btn-vimeo btn-brand mr-1 mb-1">
+                <CIcon
+                  content={cibVimeo}
+                  name="cib-vimeo"
+                />
+              </CButton>
+            </p>
```
Comment (jakxnz) [inline src/stories/buttons.stories.js:2268]: Before I spin up my local to test. Did we manage to get these buttons with icons working?

**Review by jakxnz — COMMENTED** [review]
Hey dude, looking really nice. Excited to get this in. Just a couple of questions to answer first!

**Review by Krank3n — COMMENTED** [review]
> Hey dude, looking really nice. Excited to get this in. Just a couple of questions to answer first!

Great, yes managed to get the buttons icons working in the end and have removed the header

File: src/stories/buttons.stories.js (hunk 15 above)
Comment (Krank3n) [reply src/stories/buttons.stories.js:2268]: Yea we did

File: src/js/components/Header/header.stories.js (hunk 14 above)
Comment (Krank3n) [reply src/js/components/Header/header.stories.js]: removed

**Review by jakxnz — APPROVED** [review]
Looking great

**Pull request merged**